### PR TITLE
look for colormap in rcParams['axes.prop_cycle'] (mpl 1.5+) first

### LINKED ISSUE #11614: bar-plots with wrong color using Matplotlib 1.5
Hi there,

I'm not sure if this is a pandas/matplotlib or seaborn issue...but since I upgraded to Matplotlib 1.5 using anaconda I can't get the colors of my barplots right anymore, here's an example:

In ipython notebook:

```
import pandas as pd

%matplotlib inline

import seaborn as sns
sns.set_style('whitegrid')

# Works, color as expected in nice muted blue
pd.DataFrame([1,2,3,4,5,6]).plot(linewidth=10)


# Doesn't work, getting matplotlib "eye-cancer-blue"
pd.DataFrame([1,2,3,4,5,6]).plot(kind='bar')
```

I attached a screenshot one with matplotlib 1.4.x and one with 1.5:

matplotlib_15
matplotlib_14

Again I know this might well not be a pandas issue but maybe someone has an idea where the problem might be.

### PR DESCRIPTION
**Pull request by rcarneva**
Fixes #11614

### CODE REVIEW

**Comment by TomAugspurger** [comment]
Thanks!

Could you add a test and a release note? The test should ensure that the colors of the bar patches are correct.

**Comment by rcarneva** [comment]
Added a new test with both a passed colormap and one set through mpl's rcParams. What do I need to do to add a release note?

**Comment by TomAugspurger** [comment]
For the release not add a line in `doc/source/whatsnew/v0.18.0` under the bug fixes heading.

File: pandas/tools/tests/test_plotting.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    def test_plot(self):
+        color_tuples = [(0.9, 0, 0, 1), (0, 0.9, 0, 1), (0, 0, 0.9, 1)] 
+        colormap = mpl.colors.ListedColormap(color_tuples)
+        barplot = pd.DataFrame([[1,2,3]]).plot(kind="bar", cmap=colormap)
+        assert_equal(color_tuples, [c.get_facecolor() for c in barplot.patches])
+
+    def test_rcParams(self):
+        color_tuples = [(0.9, 0, 0, 1), (0, 0.9, 0, 1), (0, 0, 0.9, 1)]
+        try:
+            mpl.rcParams["axes.prop_cycle"] = mpl.cycler("color", color_tuples)
```
Comment (TomAugspurger) [inline pandas/tools/tests/test_plotting.py]: We'll need to be careful not to modify rcParams for the rest of the tests. Can't look right now, but I think we or matplotlib have a contextmanager that handles this. @sinhrks do you remember? Also you can move this to `tpandas/tools/tests/test_graphics.py` and add it to one of the test classes in there.

**Comment by rcarneva** [comment]
I think it's good to go, though looks like I forgot to pull in the most recent changes to the release notes before adding my change... oops. Let me know if there's anything else that needs doing.

**Comment by jorisvandenbossche** [comment]
@rcarneva Can squash the commits? (http://pandas-docs.github.io/pandas-docs-travis/contributing.html#combining-commits)

**Comment by rcarneva** [comment]
Thanks for the help with this, @TomAugspurger and @jorisvandenbossche

**Comment by TomAugspurger** [comment]
I'm going to merge this now. I think it has the same problem that all of our compatibility code for mpl 1.5 has: what if the user has a prop_cycle where color isn't in the cycler? May need to look into that if it's something matplotlib tries to support.

Thanks @rcarneva!

**Pull request merged**